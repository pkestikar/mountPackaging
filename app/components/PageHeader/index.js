import React from 'react';

const PageHeader = (props) => {
    return (
        <div className="mp--page-header">
            <h1>{ props.title }</h1>
        </div>
    )
};

PageHeader.defaultProps = {
    title: ''
};

export default PageHeader;
