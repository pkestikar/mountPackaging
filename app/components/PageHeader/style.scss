.mp--page-header {
  width: 100%;
  h1 {
    text-decoration: underline;
    margin-bottom: 4rem;
    @include breakpoint(small down) {
      margin-bottom: 2rem;
    }
  }
  &.centered {
    h1 {
      text-align: center;
    }
  }
}