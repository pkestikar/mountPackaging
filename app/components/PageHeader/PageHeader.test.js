import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import PageHeader from "Components/PageHeader/index";

describe('PageHeader component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<PageHeader />);
    });

    it('renders correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a blank h1 tag with a surrounding div if no title provided', () => {
        expect(wrapper.find('h1').text()).toEqual('');
    });

    it('renders the title text provided in the title prop', () => {
        let titleText = 'About Us';
        wrapper.setProps({ title: titleText });
        expect(wrapper.find('h1').text()).toEqual(titleText);
    });
});