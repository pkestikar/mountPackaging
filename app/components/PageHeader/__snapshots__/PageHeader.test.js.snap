// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component renders correctly 1`] = `
<div
  className="mp--page-header"
>
  <h1 />
</div>
`;
